pkg/client/add_events.go

```diff
@@ -279,22 +271,14 @@ func (client *DataSetClient) Shutdown() error {
 	client.Logger.Info("Shutting down - publishing all buffers")
 	client.publishAllBuffers()
 
```

Comment:
What do you think about putting little bit more effort here?

We have waiting at most half of the time for events. But it could happen, that we haven't spend so much time waiting. Maybe we can wait here for the buffers something like - `RetryShutdownTimeout - already spent time in the shutdown function`?

Reply:
Makes sense